src/utils/MainApi.js

```diff
@@ -0,0 +1,50 @@
+const BASE_URL = "https://api.news-explorer.mk.students.nomoreparties.sbs";
+
+class MainApi {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  getUserInfo() {
+    return customFetch(`${this._baseUrl}/users/me`, {
+      headers: this._headers,
+    });
+  }
+
+  getSavedArticles() {
+    return customFetch(`${this._baseUrl}/articles/`, {
+      headers: this._headers,
+    });
+  }
+
+  addArticle(articlesData) {
+    return customFetch(`${this._baseUrl}/articles`, {
+      headers: this._headers,
+      method: "POST",
+      body: JSON.stringify(articlesData),
+    });
+  }
+
+  deleteArticle(articleId) {
+    return customFetch(`${this._baseUrl}/articles/${articleId}`, {
+      headers: this._headers,
+      method: "DELETE",
+    });
+  }
+}
+
+const customFetch = (url, headers) =>
+  fetch(url, headers).then((res) =>
+    res.ok ? res.json() : Promise.reject(res.statusText)
```

Comment:
you need to use `checkResponse` for checking responses not to duplicate the code